src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -50,12 +51,7 @@ private ServiceAuthorizationBehavior(ServiceAuthorizationBehavior other)
                 CopyAuthorizationPoliciesAndManager(other);
             }
 
-            _isAuthorizationServiceSet = other._isAuthorizationServiceSet;
-            if (other._isAuthorizationServiceSet)
-            {
-                _authorizationService = other._authorizationService;
-            }
-
+            _serviceScopeFactory = other._serviceScopeFactory;
             _isReadOnly = other._isReadOnly;
```

Comment:
I presume _scope isn't being copied to avoid double Disposing of it?

Reply:
The property behaves as a factory method that pull an instance of IAuthorizationService from DI. To support scoped AuthorizationHandler we need to pull the service from within a IServiceScope. The _scope field is assigned when we pull the IAuthorizationService and disposed when ServiceAuthorizationBehavior is disposed, so the lifetime of IAuthorizationService is tied to the lifetime of the ServiceAuthorizationBehavior and its _scope field if any.

ServiceAuthorizationBehavior is registered as singleton in `AddServiceModelServices`, so making ServiceAuthorizationBehavior disposable is ok for instances that are pulled from DI but i just noticed there are usages of constructors. I will try to drop other usages

Reply:
I ensured ServiceAuthorizationBehavior is always pulled from DI and thus disposed in https://github.com/CoreWCF/CoreWCF/pull/948/commits/3958008c3c678fbc885b94a36361a83d27038b0a.